users.js

```diff
@@ -1449,7 +1455,10 @@ class User {
 			};
 		}
 		this.send('|updatechallenges|' + JSON.stringify({
-			challengesFrom: Object.map(this.challengesFrom, 'format'),
+			challengesFrom: Object.keys(this.challengesFrom).reduce((prev, next) => {
+				prev[next] = this.challengesFrom[next].format;
+				return prev;
+			}, {}),
```

Comment:
I'm not sure I like the idea of fancy reduce usage. Would rather have a for-in loop, I think.